game/scripts/vscripts/abilities/oaa_force_of_nature.lua

```diff
@@ -0,0 +1,116 @@
+--[[
+  OAA modified version of Nature's Prophet's Nature's Call/Force of Nature ability
+  Spawns upgraded treants for levels 5 and 6
+  Code based on Pizzalol's Spell Library recreation of the original Dota ability
+  and Valve's Lua ability example recreation
+
+  Modified by: Trildar
+  Date: 10.03.2017
+]]
+furion_force_of_nature = class ({})
+LinkLuaModifier( "modifier_treant_giant_bonus", "modifiers/modifier_treant_giant_bonus", LUA_MODIFIER_MOTION_NONE )
+
+function furion_force_of_nature:GetAOERadius()
+  return self:GetSpecialValueFor( "area_of_effect" )
+end
+
+function furion_force_of_nature:CastFilterResultLocation( target_point )
+  if IsServer() then
+    local area_of_effect = self:GetSpecialValueFor( "area_of_effect" )
+
+    if GridNav:IsNearbyTree( target_point, area_of_effect, true ) then
+      return UF_SUCCESS
+    else
+      return UF_FAIL_CUSTOM
+    end
+  end
+end
+
+function furion_force_of_nature:GetCustomCastErrorLocation( target_point )
+  return "#dota_hud_error_must_target_tree"
+end
+
+--[[
+  Author: Noya
+  Date: 25.01.2015.
+  Checks for trees, if the spell didn't target one on its radius, interrupt with a message
+]]
+-- function furion_force_of_nature:OnAbilityPhaseStart()
+--   local point = self:GetCursorPosition()
+--   local area_of_effect = self:GetSpecialValueFor( "area_of_effect" )
+
+--   if GridNav:IsNearbyTree( point, area_of_effect, true ) then
+--     --print(ability,"Trees found")
+--     return true
+--   else
+--     FireGameEvent( 'dota_hud_error_message', { reason = 80, message = "#dota_hud_error_must_target_tree" } )
+--     return false
+--   end
+-- end
```

Comment:
Derp. It's sort of an alternate way to do the "Ability Must Target Tree" cast error that's currently done using `CastFilterResultLocation` and the accompanying `GetCustomCastErrorLocation`. The commented function was what I used first cause it was adapted from the Spell Library.

Guess I should just remove that block?